rmgpy/data/thermo.py

```diff
@@ -1337,8 +1337,98 @@ def getRingGroupsFromComments(self, thermoData):
+        and extracts the thermo sources.
+
+        Returns a dictionary with keys of either 'Library', 'QM', and/or 'GAV'.
+        Commonly, species thermo are estimated using only one of these sources.
+        However, a radical can be estimated with more than one type of source, for 
+        instance a saturated library value and a GAV HBI correction, or a QM saturated value
+        and a GAV HBI correction.  
+        
+        source = {'Library': String_Name_of_Library_Used,
+                  'QM': String_of_Method_Used,
+                  'GAV': Dictionary_of_Groups_Used 
+                  }
+                  
+        The Dictionary_of_Groups_Used looks like 
+        {'groupType':[List of tuples containing (Entry, Weight)]
+        """
+        comment = species.thermo.comment
+        tokens = comment.split()
+        
+        
+        source = {}
+        
+        if comment.startswith('Thermo library'):
+            # Store name of the library source, which is the 3rd token in the comments
+            source['Library'] = tokens[2]
+            
+        elif comment.startswith('QM'):
+            # Store the level of the calculation, which is the 2nd token in the comments
+            source['QM'] = tokens[1]
+        
+        
+        # Check for group additivity contributions to the thermo in this species            
+        # These groupTypes are hardcoded and must be updated if new types are added
+        
+        # The contribution of the groups can be either additive or substracting
+        # after changes to the polycyclic algorithm
+        
+        comment = comment.replace(' + ',' +')
+        comment = comment.replace(' - ', ' -')
+        
+        groups = {}
+        groupTypes = ['group','gauche','int15','other','ring','polycyclic', 'radical']
+        
+        regex = "\((.*)\)" #only hit outermost parentheses
+        for token in tokens:
+            weight = 1  # default contribution is additive
+            if token.startswith('+'):
+                token = token[1:]
+            elif token.startswith('-'):
+                weight = -1
+                token = token[1:]
+            for groupType in groupTypes:
+                if token.startswith(groupType+'(') and token.endswith(')'):
+                    splitTokens = re.split(regex, token)
+                    groupLabel = splitTokens[1]
+                    groupEntry = self.groups[groupType].entries[groupLabel]
+                    # Use dictionary to combine into weights when necessary
+                    if not groupType in groups:
+                        groups[groupType] = {groupEntry:weight}
+                    else:
+                        if groupEntry in groups[groupType]:
+                            groups[groupType][groupEntry] += weight
+                        else:
+                            groups[groupType][groupEntry] = weight
+                    break
+            
```

Comment:
for comment containing `-`, it doesn't work; it gives ```{'group': {<Entry index=3 label="Cbf-CbCbCbf">: 4,
  <Entry index=7 label="Cb-H">: 10},
 'other': {<Entry index=0 label="R">: 14},
 'polycyclic': {<Entry index=185 label="s2_6_6_naphthalene">: 2},
 'ring': {<Entry index=96 label="Benzene">: 1}}``` instead of ```{'group': {<Entry index=3 label="Cbf-CbCbCbf">: 4,
  <Entry index=7 label="Cb-H">: 10},
 'other': {<Entry index=0 label="R">: 14},
 'polycyclic': {<Entry index=185 label="s2_6_6_naphthalene">: 2},
 'ring': {<Entry index=96 label="Benzene">: -1}}```. 

My testing comment is `Thermo group additivity estimation: group(Cbf-CbCbCbf) + other(R) + group(Cbf-CbCbCbf) + other(R) + group(Cbf-CbCbCbf) + other(R) + group(Cbf-CbCbCbf) + other(R) + group(Cb-H) + other(R) + group(Cb-H) + other(R) + group(Cb-H) + other(R) + group(Cb-H) + other(R) + group(Cb-H) + other(R) + group(Cb-H) + other(R) + group(Cb-H) + other(R) + group(Cb-H) + other(R) + group(Cb-H) + other(R) + group(Cb-H) + other(R) + polycyclic(s2_6_6_naphthalene) + polycyclic(s2_6_6_naphthalene) - ring(Benzene)"`. 

Adding `tokens = comment.split()` at line 1388 should make it work.